### DISCUSSION TITLE
Typescript issues when using the options API

### DISCUSSION THREAD

**Discussion by kd-ricardo-tavares** [vuejs/pinia#704] [Help and Questions] [answered]
When using the map helpers I get a TS2339 error when trying to access a mapped store/prop in a computed prop, they work fine when used in actions so I'm wondering if this is by design?

```typescript
const useCartStore = defineStore('cart', {
  state: (): { items: { name: string }[] } => ({
    items: [],
  }),
});

export default defineComponent({
  name: 'Test',

  computed: {
    ...mapStores(useCartStore),
    ...mapState(useCartStore, ['items']),

    // these throw a ts error
    example2() {
      return this.cartStore.items;
    },
    exampleState2() {
      return this.items;
    },
  },

  methods: {
    // these work fine
    example() {
      return this.cartStore.items;
    },
    exampleState() {
      return this.items;
    },
  },
});
```
![image](https://user-images.githubusercontent.com/68226602/136252673-ce18bd6c-2bc7-428f-92e9-448e808664a2.png)

I've also noticed that when I use the `mapStores` helper my IDE (webstorm) provides type information but when I use  `mapState` it doesn't, am I missing something?
![image](https://user-images.githubusercontent.com/68226602/136253355-818b0005-fa41-4194-8ecf-748f3daa962c.png)

**Comment by posva** [comment, marked as answer]
This is a TS limitation when using `this`. You need to type the return value: https://v3.vuejs.org/guide/typescript-support.html#annotating-return-types

**Reply by kd-ricardo-tavares** [reply]
Ah right, should have known better as I've run into this issue before. Thanks!
